src/figuresClasses.ts

```diff
@@ -1,19 +1,102 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: () => number
+}
+
+function checkSides(shape: Shape, ...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error(
+      `Can't build ${shape} with these sides: ${sides.join(', ')}`,
+    );
+  }
+
+  if (shape === Shape.Triangle) {
+    const [a, b, c] = sides.sort(
+      (prev, next) => prev - next,
+    );
+
+    if (c >= a + b) {
+      throw new Error(
+        `Can't build ${shape} with these sides: ${sides.join(', ')}`,
+      );
+    }
+  }
```

Comment:
Helper functions should be as generic as it possible, so better instead move this logic to separate function